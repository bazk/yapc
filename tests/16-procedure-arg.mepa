     INPP
     AMEM 2
     DSVS R00
R01: ENPR 1 
     AMEM 1
     DSVS R02
R02: NADA 
     CRCT 2
     CRVL 1, -4
     MULT
     ARMZ 1, -4
     CRCT 2
     CRVL 1, -4
     MULT
     ARMZ 1, 0
     CRVL 1, 0
     CRCT 1
     SOMA
     ARMZ 0, 0
     DMEM 1
     RTPR 1, 1
R00: NADA 
     CRCT 3
     ARMZ 0, 0
     CRCT 4
     ARMZ 0, 1
     CRVL 0, 1
     CHPR R01, 0
     CRVL 0, 0
     IMPR
     CRVL 0, 1
     IMPR
     CRVL 0, 0
     CHPR R01, 0
     CRVL 0, 0
     IMPR
     CRVL 0, 1
     IMPR
     CRVL 0, 1
     CRVL 0, 0
     SOMA
     CHPR R01, 0
     CRVL 0, 0
     IMPR
     CRVL 0, 1
     IMPR
     CRCT 7
     CHPR R01, 0
     CRVL 0, 0
     IMPR
     CRVL 0, 1
     IMPR
     DMEM 2
     PARA
