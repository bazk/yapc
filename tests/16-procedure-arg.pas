program exemplo3 (input, output);
var z, t :  integer;
   procedure g(t : integer);
   var x :  integer;
   begin
      t:=2*t;
      x:=2*t;
      z:=x+1
   end;
begin
   z:=3;
   t:=4;
   g(t); write (z,t);
   g(z); write (z,t);
   g(t+z); write (z,t);
   g(7); write (z,t)
end.
   