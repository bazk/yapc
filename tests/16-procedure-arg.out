1746942934294